vscode/src/format.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { ILanguageService } from "qsharp-lang";
+import * as vscode from "vscode";
+import { toVscodeRange } from "./common";
+import { EventType, sendTelemetryEvent } from "./telemetry";
+import { getRandomGuid } from "./utils";
+
+export function createFormatProvider(languageService: ILanguageService) {
+  return new QSharpFormatProvider(languageService);
+}
+
+class QSharpFormatProvider implements vscode.DocumentFormattingEditProvider {
+  constructor(public languageService: ILanguageService) {}
+
+  async provideDocumentFormattingEdits(document: vscode.TextDocument) {
+    // telemetry start format
+    const associationId = getRandomGuid();
+    sendTelemetryEvent(EventType.FormatStart, { associationId }, {});
+    const start = performance.now();
+
+    const lsEdits = await this.languageService.getFormatChanges(
+      document.uri.toString(),
+    );
+
+    if (!lsEdits) return [];
+    const edits = [];
+    for (const edit of lsEdits) {
+      const referenceRange = toVscodeRange(edit.range);
+      edits.push(new vscode.TextEdit(referenceRange, edit.newText));
+    }
```

Comment:
More concise (and a little more idiomatic) to write this using `.map`

```typescript
const edits = lsEdits.map(edit => new vscode.TextEdit(toVsCodeRange(edit.range), edit.newText))
```

Reply:
done